flowpipe/plug.py

```diff
@@ -166,11 +169,18 @@ def value(self, value):
         for plug in self.connections:
             plug.value = value
 
+    def __rshift__(self, other):
+        """Syntactic sugar for the connect() method."""
+        self.connect(other)
+
     def connect(self, plug):
         """Connect this Plug to the given InputPlug.
 
         Set both participating Plugs dirty.
         """
+        if not isinstance(plug, self.accepted_plugs):
```

Comment:
I think raising these errors is more pythonic, as otherwise trying to connect two output plugs just fails silently, not creating a connection.

I was about 50:50 on whether it should be a ValueError or TypeError.